### ISSUE TITLE
Force gzip transfer in HTTP

### ISSUE THREAD

**Issue by jaakla** [CartoDB/mobile-sdk#212]
From the Wireshark iOS traffic it seems that vector tile API traffic does not use gzip encoding/compression ?
```
GET /vectortiles/carto.streets/v1/11/1022/680.mvt?appToken=REDACTED HTTP/1.1
Host: tiles.basemaps.cartocdn.com
Accept: */*
User-Agent: libg-cap/1.830.765 CFNetwork/894 Darwin/17.4.0
Referer: app://com.epam.cap.gb.vm.internal
Connection: keep-alive

HTTP/1.1 200 OK
Content-Encoding: gzip
Access-Control-Allow-Credentials: true
Access-Control-Allow-Origin: *
Cache-Control: max-age=604800
Content-Type: application/x-protobuf
Date: Mon, 23 Apr 2018 14:55:43 GMT
Last-Modified: Thu, 29 Mar 2018 09:33:22 GMT
Server: nginx
Strict-Transport-Security: max-age=63072000; includeSubdomains
Surrogate-Control: max-age=604800
Surrogate-Keys: omt-vector-z-11 omt-vector-tile-11_1022_680
X-Content-Type-Options: nosniff
X-Frame-Options: DENY
X-Origin-Server: basemaps-01-lb2-Eastus
X-Powered-By: Express
Transfer-Encoding: chunked

...x.
.water(. ..class".
...
```
same for CARTO Maps API:
```
GET /lgi-hotspot-admin/api/v1/map/lgi-hotspot-admin@cc3aa909@b60e3bc810b4f08f5606ea3768fa2349:1524460423649/1,2,3/16/33693/21611.mvt HTTP/1.1
Host: gewd.cartocdn.com
Accept: */*
User-Agent: libg-cap/1.830.765 CFNetwork/894 Darwin/17.4.0
Connection: keep-alive

HTTP/1.1 200 OK
Server: openresty
Content-Type: application/x-protobuf
Access-Control-Allow-Origin: *
Access-Control-Allow-Headers: X-Requested-With, X-Prototype-Version, X-CSRF-Token
x-tilelive-contains-data: true
Cache-Control: public,max-age=31536000
Last-Modified: Mon, 23 Apr 2018 05:13:43 GMT
Content-Length: 1204
Accept-Ranges: bytes
Date: Mon, 23 Apr 2018 14:55:17 GMT
Via: 1.1 varnish
Age: 0
Connection: keep-alive
X-Served-By: cache-ams4431-AMS
X-Cache: MISS
X-Cache-Hits: 0
X-Timer: S1524495318.771430,VS0,VE53
Vary: Accept-Encoding, Authorization

..	x.
$c73b9283-96eb-4981-8074-728700d4675f(. ..abbrev".
.Neth..
abbrev_len".....@..adm0_a3".
...
```

@mtehver can you check it? Or should app set `Accept-Encoding: gzip` request header for both APIs itself?

**Comment by mtehver** [comment]
@jaakla Will investigate this. From what I have gathered NSURLConnection API should do this automatically, but SDK uses CFNetwork, which is somewhat lower level. BTW, it is possible that similar issue affects Android also.

**Comment by jaakla** [comment]
If by default the map HTTP traffic is not compressed, then I believe it could be resolved in two ways:
a) update SDK - have default compression on for all HTTP API-s, I don't see use case where this is not preferred or where it could break compatibility
b) update app to add folloiwng lines to enable compression with existing SDK.
```java
// define custom header
StringMap headers = new StringMap();
headers.set("Accept-Encoding","gzip");
// following for every Layer:
((HTTPTileDataSource) layer.getDataSource()).setHTTPHeaders(headers); 
```
It seems to affect both iOS and Android.

**Comment by jaakla** [comment]
Update: on Android gzip seems to be turned on without need to change code. My test session: about 12 clicks (map views) generated 32 tile requests, 600K basemap data and 900 user tile data in UK, with a lot of points. This traffic is ok.

On iOS cannot get the traffic with CharlesApp, as it seems that CFNetwork also bypasses phone-configured HTTP proxy. This is probably rarely used nowadays, except for developer sniffing traffic, but could be issue for some corporate networks.

**Comment by jaakla** [comment]
Implemented in TCP level now, bigger fix is #217
